src/grt/src/GlobalRouter.cpp

```diff
@@ -3776,6 +3802,139 @@ std::vector<odb::dbNet*> GlobalRouter::getNetsToRoute()
   return nets_to_route_;
 }
 
+void GlobalRouter::mergeNetsRouting(odb::dbNet* db_net1, odb::dbNet* db_net2)
+{
+  Net* net1 = db_net_map_[db_net1];
+  Net* net2 = db_net_map_[db_net2];
+  // Do not merge the routing if the survivor net is already dirty
+  if (!net1->isDirtyNet()) {
+    GRoute& net1_route = routes_[db_net1];
+    GRoute& net2_route = routes_[db_net2];
+    net1_route.insert(net1_route.end(), net2_route.begin(), net2_route.end());
+    connectRouting(net1_route, db_net1, db_net2);
+    net1->setSkipIncremental(true);
+    net2->setSkipIncremental(true);
+  }
+}
+
+void GlobalRouter::connectRouting(GRoute& route,
+                                  odb::dbNet* db_net1,
+                                  odb::dbNet* db_net2)
+{
+  Net* net1 = db_net_map_[db_net1];
+  Net* net2 = db_net_map_[db_net2];
+
+  // find the pin positions in the buffer that connects the two nets
+  odb::Point pin_pos1;
+  odb::Point pin_pos2;
+  findBufferPinPostions(net1, net2, pin_pos1, pin_pos2);
+
+  if (pin_pos1 != pin_pos2) {
+    const int layer1 = findTopLayerOverPosition(pin_pos1, route);
+    const int layer2 = findTopLayerOverPosition(pin_pos2, route);
+    std::vector<GSegment> connection
+        = createConnectionForPositions(pin_pos1, pin_pos2, layer1, layer2);
+    route.insert(route.end(), connection.begin(), connection.end());
+  }
+}
+
+void GlobalRouter::findBufferPinPostions(Net* net1,
+                                         Net* net2,
+                                         odb::Point& pin_pos1,
+                                         odb::Point& pin_pos2)
+{
+  for (const Pin& pin1 : net1->getPins()) {
+    if (!pin1.isPort()) {
+      for (const Pin& pin2 : net2->getPins()) {
+        if (!pin2.isPort()) {
```

Comment:
This could be more efficient though I'm not sure it matters.

Reply:
I found that way easier because I already have access to both pin positions on the grid. I could iterate only through the pins of one net and find the buffer that connects them, but I would need to compute the position on grid of the pins.